File: src/main/resources/output/IFC4X3_js/Entities/IfcMaterialDefinition/IfcMaterialProfile/IfcMaterialProfile.js
class IfcMaterialProfile extends IfcMaterialDefinition {
    constructor() {
        /** @type {IFCLABEL} */
        this.Name = null;
        /** @type {IFCTEXT} */
        this.Description = null;
        /** @type {IFCMATERIAL} */
        this.Material = null;
        /** @type {IFCPROFILEDEF} */
        this.Profile = null;
        /** @type {IFCINTEGER} */
        this.Priority = null;
        /** @type {IFCLABEL} */
        this.Category = null;
    }

    // === EXTENDED BY ===
    // IfcMaterialProfileWithOffsets

    // === INVERSE CLAUSES ===
    // ToMaterialProfileSet : IfcMaterialProfileSet FOR MaterialProfiles

    // === WHERE CLAUSES ===
    // NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}
}

File: src/main/resources/output/IFC4X3_java/Entities/IfcMaterialDefinition/IfcMaterialProfile/IfcMaterialProfile.java
public class IfcMaterialProfile extends IfcMaterialDefinition {
    public IfcLabel Name;
    public IfcText Description;
    public IfcMaterial Material;
    public IfcProfileDef Profile;
    public IfcInteger Priority;
    public IfcLabel Category;

    // === EXTENDED BY ===
    // IfcMaterialProfileWithOffsets

    // === INVERSE CLAUSES ===
    // ToMaterialProfileSet : IfcMaterialProfileSet FOR MaterialProfiles

    // === WHERE CLAUSES ===
    // NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}
}
